gadget-container/entrypoint.sh

```diff
@@ -56,21 +59,53 @@ if grep -q '^1:name=systemd:.*/crio-[0-9a-f]*\.scope$' /proc/self/cgroup > /dev/
     CRIO=1
 fi
 
-if grep -q '^ID="rhcos"$' /host/etc/os-release > /dev/null ; then
-  if [ ! -d "/host/usr/src/kernels/$(uname -r)" ] ; then
-    echo "Fetching kernel-devel from CentOS 8."
-    REPO=http://mirror.centos.org/centos/8/BaseOS/x86_64/os/Packages/
-    RPM=kernel-devel-$(uname -r).rpm
-    RPMDIR=/opt/gadget-kernel/
-    RPMHOSTDIR=/host${RPMDIR}
-    mkdir -p $RPMHOSTDIR/usr/src/kernels/
-    test -r $RPMHOSTDIR/$RPM || \
-        curl -fsSLo $RPMHOSTDIR/$RPM $REPO/$RPM
-    test -r $RPMHOSTDIR/usr/src/kernels/`uname -r`/.config || \
-        chroot /host sh -c "cd $RPMDIR && rpm2cpio $RPM | cpio -i"
-    test ! -L /usr/src || rm -f /usr/src
-    mkdir -p /usr/src/kernels/`uname -r`/
-    mount --bind $RPMHOSTDIR/usr/src/kernels/`uname -r` /usr/src/kernels/`uname -r`
+if [ "$ID" = "rhcos" ] && [ ! -d "/host/usr/src/kernels/$KERNEL" ]; then
```

Comment:
Suggested comment:
```
# In the gadget image, /usr/src is a symlink to /host/usr/src (see gadget.Dockerfile).
# If the kernel headers are already available on the host via the symlink, no need to download them.
```

And I ponder about testing directly the container path, since bcc scripts will later access it via that path:
```suggestion
if [ "$ID" = "rhcos" ] && [ ! -d "/usr/src/kernels/$KERNEL" ]; then
```